File: lib/models/invoice_image_status.dart
enum InvoiceImageStatus {
  uploaded,
  ocrInProgress,
  ocrError,
  ocrNoText,
  ocrFinished,
  analyzeInProgress,
  analyzeError,
  analyzeFinished,
}

extension InvoiceImageStatusX on InvoiceImageStatus {
  String get asString => toString().split('.').last;
  static InvoiceImageStatus fromString(String value) =>
      InvoiceImageStatus.values.firstWhere(
        (e) => e.asString == value,
        orElse: () => InvoiceImageStatus.uploaded,
      );
}

File: functions/src/invoice-capture/invoice_image_status.ts
export type InvoiceImageStatus =
  | "uploaded"
  | "ocrStart"
  | "ocrInProgress"
  | "ocrError"
  | "ocrNoText"
  | "ocrFinished"
  | "analyzeStart"
  | "analyzeInProgress"
  | "analyzeError"
  | "analyzeFinished";

export const allowedTransitions: Record<InvoiceImageStatus, InvoiceImageStatus[]> = {
  uploaded: ["ocrStart"],
  ocrStart: ["ocrInProgress"],
  ocrInProgress: ["ocrError", "ocrNoText", "ocrFinished"],
  ocrError: [],
  ocrNoText: ["analyzeStart"],
  ocrFinished: ["analyzeStart"],
  analyzeStart: ["analyzeInProgress"],
  analyzeInProgress: ["analyzeError", "analyzeFinished"],
  analyzeError: [],
  analyzeFinished: [],
}; 